retrieval/discovery/marathon/marathon.go

```diff
@@ -47,10 +48,32 @@ const appListPath string = "/v2/apps/?embed=apps.tasks"
 
 // Discovery provides service discovery based on a Marathon instance.
 type Discovery struct {
-	Servers         []string
-	RefreshInterval time.Duration
+	client          *http.Client
+	servers         []string
+	refreshInterval time.Duration
 	lastRefresh     map[string]*config.TargetGroup
-	Client          AppListClient
+	appsClient      AppListClient
+}
+
+// Initialize sets up the discovery for usage.
+func NewDiscovery(conf *config.MarathonSDConfig) (*Discovery, error) {
+	tls, err := httputil.NewTLSConfig(conf.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
```

Comment:
@juliusv  Yes, I completely agree with you but can I add this feature with another merge request?
Which timeout and default value would you like to have for Marathon? `net.Dialer.Timeout`?